### ISSUE TITLE
PoolManager.urlopen fails with TypeError for http connection if the PoolManager is instantiated with server_hostname

### ISSUE THREAD

**Issue by kajinamit** [urllib3/urllib3#2534]
### Subject

PoolManager.urlopen fails with TypeError for http connection if the PoolManager is instantiated with server_hostname

### Environment

```
>>> import platform
>>> import urllib3
>>> 
>>> print("OS", platform.platform())
OS Linux-5.11.22-100.fc32.x86_64-x86_64-with-glibc2.2.5
>>> print("Python", platform.python_version())
Python 3.8.10
>>> print("urllib3", urllib3.__version__)
urllib3 1.26.8

```

### Steps to Reproduce

Create a PoolManager with server_hostname and call urlopen method of the instance.
```
>>> import urllib3
>>> manager = urllib3.PoolManager(server_hostname='foo')
>>> manager.urlopen('GET', 'http://127.0.0.1') 
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/home/tkajinam/virtualenv/foo/lib/python3.8/site-packages/urllib3/poolmanager.py", line 375, in urlopen
    response = conn.urlopen(method, u.request_uri, **kw)
  File "/home/tkajinam/virtualenv/foo/lib/python3.8/site-packages/urllib3/connectionpool.py", line 692, in urlopen
    conn = self._get_conn(timeout=pool_timeout)
  File "/home/tkajinam/virtualenv/foo/lib/python3.8/site-packages/urllib3/connectionpool.py", line 281, in _get_conn
    return conn or self._new_conn()
  File "/home/tkajinam/virtualenv/foo/lib/python3.8/site-packages/urllib3/connectionpool.py", line 235, in _new_conn
    conn = self.ConnectionCls(
  File "/home/tkajinam/virtualenv/foo/lib/python3.8/site-packages/urllib3/connection.py", line 130, in __init__
    _HTTPConnection.__init__(self, *args, **kw)
TypeError: __init__() got an unexpected keyword argument 'server_hostname'
```

### Expected Behavior

TypeError should not be raised and a request is sent

### Actual Behavior

TypeError is raised and a request is not sent

**Comment by kajinamit** [comment]
FYI: This was initially reported in https://github.com/cdent/gabbi/issues/309 .

**Comment by kajinamit** [comment]
Just adding some more results...

TypeError is not raised when I use https. This is because HTTPSConnection DOES accept the server_hostname argument.
(Connection refused is expected because I haven't start up any process listening on that port)
```
>>> import urllib3
>>> manager = urllib3.PoolManager(server_hostname='foo')
>>> manager.urlopen('GET', 'https://127.0.0.1') 
Traceback (most recent call last):
  File "/home/tkajinam/virtualenv/foo/lib/python3.8/site-packages/urllib3/connection.py", line 174, in _new_conn
    conn = connection.create_connection(
  File "/home/tkajinam/virtualenv/foo/lib/python3.8/site-packages/urllib3/util/connection.py", line 95, in create_connection
    raise err
  File "/home/tkajinam/virtualenv/foo/lib/python3.8/site-packages/urllib3/util/connection.py", line 85, in create_connection
    sock.connect(sa)
ConnectionRefusedError: [Errno 111] Connection refused

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/home/tkajinam/virtualenv/foo/lib/python3.8/site-packages/urllib3/connectionpool.py", line 703, in urlopen
    httplib_response = self._make_request(
  File "/home/tkajinam/virtualenv/foo/lib/python3.8/site-packages/urllib3/connectionpool.py", line 386, in _make_request
    self._validate_conn(conn)
  File "/home/tkajinam/virtualenv/foo/lib/python3.8/site-packages/urllib3/connectionpool.py", line 1040, in _validate_conn
    conn.connect()
  File "/home/tkajinam/virtualenv/foo/lib/python3.8/site-packages/urllib3/connection.py", line 358, in connect
    conn = self._new_conn()
  File "/home/tkajinam/virtualenv/foo/lib/python3.8/site-packages/urllib3/connection.py", line 186, in _new_conn
    raise NewConnectionError(
urllib3.exceptions.NewConnectionError: <urllib3.connection.HTTPSConnection object at 0x7f41c63d4130>: Failed to establish a new connection: [Errno 111] Connection refused

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/home/tkajinam/virtualenv/foo/lib/python3.8/site-packages/urllib3/poolmanager.py", line 375, in urlopen
    response = conn.urlopen(method, u.request_uri, **kw)
  File "/home/tkajinam/virtualenv/foo/lib/python3.8/site-packages/urllib3/connectionpool.py", line 813, in urlopen
    return self.urlopen(
  File "/home/tkajinam/virtualenv/foo/lib/python3.8/site-packages/urllib3/connectionpool.py", line 813, in urlopen
    return self.urlopen(
  File "/home/tkajinam/virtualenv/foo/lib/python3.8/site-packages/urllib3/connectionpool.py", line 813, in urlopen
    return self.urlopen(
  File "/home/tkajinam/virtualenv/foo/lib/python3.8/site-packages/urllib3/connectionpool.py", line 785, in urlopen
    retries = retries.increment(
  File "/home/tkajinam/virtualenv/foo/lib/python3.8/site-packages/urllib3/util/retry.py", line 592, in increment
    raise MaxRetryError(_pool, url, error or ResponseError(cause))
urllib3.exceptions.MaxRetryError: HTTPSConnectionPool(host='127.0.0.1', port=443): Max retries exceeded with url: / (Caused by NewConnectionError('<urllib3.connection.HTTPSConnection object at 0x7f41c63d4130>: Failed to establish a new connection: [Errno 111] Connection refused'))
```

If create a PoolManager without server_hostname, urlopen works for both http and https without TypeError.
(again Connection refused error is expected)
```
>>> manager = urllib3.PoolManager()
>>> manager.urlopen('GET', 'http://127.0.0.1') 
Traceback (most recent call last):
  File "/home/tkajinam/virtualenv/foo/lib/python3.8/site-packages/urllib3/connection.py", line 174, in _new_conn
    conn = connection.create_connection(
  File "/home/tkajinam/virtualenv/foo/lib/python3.8/site-packages/urllib3/util/connection.py", line 95, in create_connection
    raise err
  File "/home/tkajinam/virtualenv/foo/lib/python3.8/site-packages/urllib3/util/connection.py", line 85, in create_connection
    sock.connect(sa)
ConnectionRefusedError: [Errno 111] Connection refused

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/home/tkajinam/virtualenv/foo/lib/python3.8/site-packages/urllib3/connectionpool.py", line 703, in urlopen
    httplib_response = self._make_request(
  File "/home/tkajinam/virtualenv/foo/lib/python3.8/site-packages/urllib3/connectionpool.py", line 398, in _make_request
    conn.request(method, url, **httplib_request_kw)
  File "/home/tkajinam/virtualenv/foo/lib/python3.8/site-packages/urllib3/connection.py", line 239, in request
    super(HTTPConnection, self).request(method, url, body=body, headers=headers)
  File "/usr/lib64/python3.8/http/client.py", line 1252, in request
    self._send_request(method, url, body, headers, encode_chunked)
  File "/usr/lib64/python3.8/http/client.py", line 1298, in _send_request
    self.endheaders(body, encode_chunked=encode_chunked)
  File "/usr/lib64/python3.8/http/client.py", line 1247, in endheaders
    self._send_output(message_body, encode_chunked=encode_chunked)
  File "/usr/lib64/python3.8/http/client.py", line 1007, in _send_output
    self.send(msg)
  File "/usr/lib64/python3.8/http/client.py", line 947, in send
    self.connect()
  File "/home/tkajinam/virtualenv/foo/lib/python3.8/site-packages/urllib3/connection.py", line 205, in connect
    conn = self._new_conn()
  File "/home/tkajinam/virtualenv/foo/lib/python3.8/site-packages/urllib3/connection.py", line 186, in _new_conn
    raise NewConnectionError(
urllib3.exceptions.NewConnectionError: <urllib3.connection.HTTPConnection object at 0x7f41c625c2e0>: Failed to establish a new connection: [Errno 111] Connection refused

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/home/tkajinam/virtualenv/foo/lib/python3.8/site-packages/urllib3/poolmanager.py", line 375, in urlopen
    response = conn.urlopen(method, u.request_uri, **kw)
  File "/home/tkajinam/virtualenv/foo/lib/python3.8/site-packages/urllib3/connectionpool.py", line 813, in urlopen
    return self.urlopen(
  File "/home/tkajinam/virtualenv/foo/lib/python3.8/site-packages/urllib3/connectionpool.py", line 813, in urlopen
    return self.urlopen(
  File "/home/tkajinam/virtualenv/foo/lib/python3.8/site-packages/urllib3/connectionpool.py", line 813, in urlopen
    return self.urlopen(
  File "/home/tkajinam/virtualenv/foo/lib/python3.8/site-packages/urllib3/connectionpool.py", line 785, in urlopen
    retries = retries.increment(
  File "/home/tkajinam/virtualenv/foo/lib/python3.8/site-packages/urllib3/util/retry.py", line 592, in increment
    raise MaxRetryError(_pool, url, error or ResponseError(cause))
urllib3.exceptions.MaxRetryError: HTTPConnectionPool(host='127.0.0.1', port=80): Max retries exceeded with url: / (Caused by NewConnectionError('<urllib3.connection.HTTPConnection object at 0x7f41c625c2e0>: Failed to establish a new connection: [Errno 111] Connection refused'))
>>> manager.urlopen('GET', 'https://127.0.0.1') 
Traceback (most recent call last):
  File "/home/tkajinam/virtualenv/foo/lib/python3.8/site-packages/urllib3/connection.py", line 174, in _new_conn
    conn = connection.create_connection(
  File "/home/tkajinam/virtualenv/foo/lib/python3.8/site-packages/urllib3/util/connection.py", line 95, in create_connection
    raise err
  File "/home/tkajinam/virtualenv/foo/lib/python3.8/site-packages/urllib3/util/connection.py", line 85, in create_connection
    sock.connect(sa)
ConnectionRefusedError: [Errno 111] Connection refused

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/home/tkajinam/virtualenv/foo/lib/python3.8/site-packages/urllib3/connectionpool.py", line 703, in urlopen
    httplib_response = self._make_request(
  File "/home/tkajinam/virtualenv/foo/lib/python3.8/site-packages/urllib3/connectionpool.py", line 386, in _make_request
    self._validate_conn(conn)
  File "/home/tkajinam/virtualenv/foo/lib/python3.8/site-packages/urllib3/connectionpool.py", line 1040, in _validate_conn
    conn.connect()
  File "/home/tkajinam/virtualenv/foo/lib/python3.8/site-packages/urllib3/connection.py", line 358, in connect
    conn = self._new_conn()
  File "/home/tkajinam/virtualenv/foo/lib/python3.8/site-packages/urllib3/connection.py", line 186, in _new_conn
    raise NewConnectionError(
urllib3.exceptions.NewConnectionError: <urllib3.connection.HTTPSConnection object at 0x7f41c625ca30>: Failed to establish a new connection: [Errno 111] Connection refused

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/home/tkajinam/virtualenv/foo/lib/python3.8/site-packages/urllib3/poolmanager.py", line 375, in urlopen
    response = conn.urlopen(method, u.request_uri, **kw)
  File "/home/tkajinam/virtualenv/foo/lib/python3.8/site-packages/urllib3/connectionpool.py", line 813, in urlopen
    return self.urlopen(
  File "/home/tkajinam/virtualenv/foo/lib/python3.8/site-packages/urllib3/connectionpool.py", line 813, in urlopen
    return self.urlopen(
  File "/home/tkajinam/virtualenv/foo/lib/python3.8/site-packages/urllib3/connectionpool.py", line 813, in urlopen
    return self.urlopen(
  File "/home/tkajinam/virtualenv/foo/lib/python3.8/site-packages/urllib3/connectionpool.py", line 785, in urlopen
    retries = retries.increment(
  File "/home/tkajinam/virtualenv/foo/lib/python3.8/site-packages/urllib3/util/retry.py", line 592, in increment
    raise MaxRetryError(_pool, url, error or ResponseError(cause))
urllib3.exceptions.MaxRetryError: HTTPSConnectionPool(host='127.0.0.1', port=443): Max retries exceeded with url: / (Caused by NewConnectionError('<urllib3.connection.HTTPSConnection object at 0x7f41c625ca30>: Failed to establish a new connection: [Errno 111] Connection refused'))
```

**Comment by cdent** [comment]
As far as I can tell a new version of urllib3 including this fix has not yet been released. 1.26.8 is the latest on pypi. Are there plans for a new release?

**Comment by scottwallacesh** [comment]
The released [v1.26.9](https://github.com/urllib3/urllib3/releases/tag/1.26.9) yesterday!  Your message must have prodded them.
